src/cc_catalog_airflow/dags/provider_api_scripts/common/storage/image.py

```diff
@@ -307,19 +333,35 @@ def _flush_buffer(self):
         if buffer_length > 0:
             logger.info(
                 'Writing {} lines from buffer to disk.'
-                .format(buffer_length)
+                  .format(buffer_length)
             )
             with open(self._OUTPUT_PATH, 'a') as f:
                 f.writelines(self._image_buffer)
                 self._image_buffer = []
                 logger.debug(
                     'Total Images Processed so far:  {}'
-                    .format(self._total_images)
+                      .format(self._total_images)
                 )
         else:
             logger.debug('Empty buffer!  Nothing to write.')
         return buffer_length
 
+    def _tag_blacklisted(self, tag):
+        """
+        Tag is banned or contains a banned substring.
+        :param tag: the tag to be verified against the blacklist
+        :return: true if tag is blacklisted, else returns false
+        """
+        if type(tag) == dict:  # check if the tag is already enriched
+            return False
```

Comment:
For this branch, you still need to check within the tag to see if `tag.get('name' ) is blacklisted.

Reply:
Added that step